Add collective social links support

hdiniz opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/5097
Require https://github.com/opencollective/opencollective-api/pull/8316

# Description

Adds option for accounts to set a number of social links to other sites.

# Screenshots

Setting
![image](https://user-images.githubusercontent.com/5448927/208414484-3fdc6820-e718-4a97-bd3c-0dae07f5a3c3.png)

Displayed in collective hero.
![image](https://user-images.githubusercontent.com/5448927/208414560-363a7c26-b379-4ec0-966d-e79872ff5fae.png)

Link is verifiable using relMe, for example in Mastodon using local ngrok url as test:
![image](https://user-images.githubusercontent.com/5448927/208416499-13c45062-9139-40ac-be8d-baa6a5f81f63.png)

znarf commented on components/collective-page/hero/HeroSocialLinks.tsx:
```diff
@@ -0,0 +1,61 @@
+import StyledLink from '../../StyledLink';
+import StyledRoundButton from '../../StyledRoundButton';
+
+function iconForSocialLinkType(type: SocialLinkType) {
```
Nice!

Betree reviewed (commented):
:ice_cream: The space here seems too small

![image](https://user-images.githubusercontent.com/1556356/209649042-a2cce129-0ff6-426a-a3a9-e55485f80a5f.png)

Betree commented:
Regarding UX, it would be great to auto-detect the link type (whenever possible). You can very easily end up with a Facebook link categorized as "Discord":

![Peek 2022-12-27 10-58](https://user-images.githubusercontent.com/1556356/209649246-4ae54479-18a8-4936-86bb-3c39fd58ad54.gif)

Here's what I would love to see:
- Paste your link
- If we detect a known service (Discord, Facebook, etc) then we can directly set the type and disable the input
- Otherwise, just let people pick between services that cannot be auto-detected: Website, Mattermost, Mastodon, GItlab, etc

We can iterate on this in a follow-up.

Betree commented on components/edit-collective/SocialLinksFormField.tsx:
```diff
@@ -0,0 +1,270 @@
+          {(value || []).map((socialLink, i) => {
+            return (
+              <SocialLinkItem
+                // eslint-disable-next-line react/no-array-index-key
+                key={i}
```
Can't indexing by `i` cause issues since we allow re-ordering? I wasn't able to spot an issue, but out of safety I would recommend:

```suggestion
                key={`${i}-{socialLink.type}-${socialLink.url}`}
```

hdiniz replied:
Keying with the url does not work well in this case, it causes unresponsive typing as the component rerenders on every key stroke of the url input.

Betree reviewed (changes requested):
Looks and feels great! My only blocker it that it breaks the layout of the settings page on mobile (see width < 455px). Also added some more general feedback below.

hdiniz commented:
Fixed layout on mobile width, added placeholder and validation of url input.

![image](https://user-images.githubusercontent.com/5448927/210554117-20f4216f-1dd8-4207-8aa8-63472d5a619b.png)

Betree reviewed (approved):
Perfect!
